backend/workflow_manager/workflow/workflow_helper.py

```diff
@@ -531,13 +534,15 @@ def execute_bin(
                     execution_mode=execution_mode,
                     hash_values_of_files=hash_values,
                 )
-            except Exception as e:
-                return ExecutionResponse(
-                    workflow_id=workflow_id,
-                    execution_id=execution_id,
-                    execution_status=ExecutionStatus.ERROR.value,
-                    error=str(e),
+            except Exception as error:
+                error_message = traceback.format_exc()
+                logger.error(
+                    f"Error executing execution {workflow_execution}: {error_message}"
+                )
+                WorkflowExecutionServiceHelper.update_execution_err(
+                    execution_id, str(error)
                 )
+                raise
```

Comment:
@muhammad-ali-e  
```suggestion
                raise error
```

Reply:
Both are okay. Raise will raise the same exception it handled